File: libs/csharp/common/src/Core/Exceptions/NotImplementedParsingException.cs
namespace Crosslight.Core.Exceptions;

/// <summary>
/// Parsing for this node configuration has not yet been implemented.
/// </summary>
public class NotImplementedParsingException : ParsingException
{
    public NotImplementedParsingException(Node errorNode)
        : base(errorNode)
    {
    }

    public NotImplementedParsingException(string message, Node errorNode)
        : base(message, errorNode)
    {
    }

    public NotImplementedParsingException(string message, Exception inner, Node errorNode)
        : base(message, inner, errorNode)
    {
    }
}

File: libs/cpp/lang/src/exceptions/not_implemented_parsing_exception.cpp
#include "lang/exceptions/not_implemented_parsing_exception.hpp"

using namespace cl::lang::exceptions;

not_implemented_parsing_exception::not_implemented_parsing_exception(const std::string& what_arg)
    : parsing_exception{what_arg} {}

not_implemented_parsing_exception::not_implemented_parsing_exception(const char* what_arg)
    : parsing_exception{what_arg} {}

const char*
not_implemented_parsing_exception::what() const noexcept {
    return _message.c_str();
}
